### ISSUE TITLE
private_role_vars=yes prevents to import roles using vars from the parent role

### ISSUE THREAD

**Issue by kvaps** [ansible/ansible#57477]
##### SUMMARY
`private_role_vars=yes` prevents to import roles using vars from the parent role

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
import_role

##### ANSIBLE VERSION

```paste below
ansible 2.8.0
  config file = /tmp/bug-57477/ansible.cfg
  configured module search path = ['/home/kvaps/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.3 (default, Mar 26 2019, 21:43:19) [GCC 8.2.1 20181127]
```

##### CONFIGURATION
```paste below
DEFAULT_PRIVATE_ROLE_VARS(/tmp/bug-57477/ansible.cfg) = True
```

##### OS / ENVIRONMENT
```
Arch Linux
```

##### STEPS TO REPRODUCE

```bash
mkdir /tmp/bug-57477
cd /tmp/bug-57477
mkdir -p roles/{go_walk,print_string}/tasks

cat > roles/print_string/tasks/main.yaml <<EOT
- debug:
    var: string
EOT

cat > roles/go_walk/tasks/main.yaml <<EOT
- import_role:
    name: print_string
  vars:
    string: "{{ user }} went for a walk"
EOT

cat > plabook.yaml <<EOT
- hosts: localhost
  gather_facts: false
  roles:

  - name: go_walk
    vars:
      user: "Leo Tolstoi"
EOT

cat > ansible.cfg <<EOT
[defaults]
private_role_vars   = yes
EOT

ansible-playbook plabook.yaml
```

##### EXPECTED RESULTS

**print_string** role should inherit var `string="Leo Tolstoi went for a walk"` because this var is intentionally specified for the **import_role** task
```
ok: [localhost] => {
    "string": "Leo Tolstoi went for a walk"
}
```

##### ACTUAL RESULTS

**print_string**  does not see var `string` because it contains variable from the parent role
```
ok: [localhost] => {
    "string": "VARIABLE IS NOT DEFINED!"
}
```

**Comment by kvaps** [comment]
Workaround to avoid this behavior is to assign this variables before using **set_fact** module and remove after:
```bash
cat > roles/go_walk/tasks/main.yaml <<EOT
- block:
  - set_fact:
      user: "{{ user }}"
  - include_role:
      name: print_string
    vars:
      string: "{{ user }} went for a walk"
  always:
  - set_fact:
      user: Null
EOT
```

**Comment by sivel** [comment]
This issue was resolved in bd03fa811b61528b242d5ef4e0f0f9dc580d2d6f and included in v2.12

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.libera.chat
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
